tests/GridFS/BucketFunctionalTest.php

```diff
@@ -708,6 +711,21 @@ public function testExistingIndexIsReused()
         $this->assertIndexNotExists($this->chunksCollection->getCollectionName(), 'files_id_1_n_1');
     }
 
+    public function testDanglingOpenWritableStream()
+    {
+        $path = __DIR__ . '/../../vendor/autoload.php';
+        @exec(
+            <<<CMD
+php -r "require '$path'; class_exists(MongoDB\Operation\FindOne::class); \\\$stream = (new MongoDB\Client)->test->selectGridFSBucket()->openUploadStream('filename', ['disableMD5' => true]);" 2>&1
+CMD,
+            $output
+        );
+
+        $output = implode(PHP_EOL, $output);
+
+        $this->assertSame('', $output);
```

Comment:
I assume we don't care about verifying the exit code (not that `exec` can do that anyway), as a segfault would _always_ product output?

Separate question: would we need to skip this test on Windows platforms?

Reply:
Yes, we always assume there will be some output, but I'll check if we can also assert against the exit code to be on the safe side.

Regarding Windows, skipping that would make sense.

Reply:
Return code is now checked as well, and we skip the test on windows.

Reply:
Oh, I missed that `exec()` _does_ capture the exit code as its third argument. Nice.